src/main/java/gov/nci/ppe/services/impl/UserServiceImpl.java

```diff
@@ -677,6 +678,19 @@ private String sendEmailToParticipantAfterCRCWithdrawsPatient(String firstName,
 				patientOptional = insertNewPatientDetailsFromOpen(newPatient);
 				newUsersList.add(patientOptional.get());
 			}else {
+				// If the patient exists, check for any changes in the relationship with Providers and CRC
+				Participant patient = (Participant)patientOptional.get();
+				Set<Provider> existingProviders = patient.getProviders();
+				CRC existingCRC = patient.getCRC();
+				if(existingProviders.size() != providerSet.size() || !providerSet.containsAll(existingProviders)) {
+					patient.setProviders(providerSet);
+				}
+				// Check if the CRC remains unchanged.
+				if(existingCRC.getOpenCtepID() != crc.getOpenCtepID()) {
+					patient.setCRC(crc);
+				}
+				
```

Comment:
Raise appropriate audit events. A separate one for CRC change, and one for Provider change.

Reply:
Implemented the changes